cpp/tests/io/experimental/hybrid_scan_filters_test.cpp

```diff
@@ -248,76 +247,66 @@ TYPED_TEST(PageFilteringWithPageIndexStats, FilterPagesWithPageIndexStats)
-
-      // Count the number of pages that survive the page index filter
-      auto const num_pages_after_page_index_filter =
-        std::accumulate(data_page_mask.begin(),
-                        data_page_mask.end(),
-                        cudf::size_type{0},
-                        [](auto sum, auto const& page_mask) {
-                          return sum + std::count(page_mask.cbegin(), page_mask.cend(), true);
-                        });
-      EXPECT_EQ(num_pages_after_page_index_filter, expected_num_pages_after_page_index_filter);
-    };
+  auto const test_filter_data_pages_with_stats = [&](cudf::ast::operation const& filter_expression,
+                                                     cudf::size_type const expected_surviving_rows,
+                                                     rmm::cuda_stream_view stream =
+                                                       cudf::get_default_stream(),
+                                                     rmm::device_async_resource_ref mr =
+                                                       cudf::get_current_device_resource_ref()) {
+    // Create reader options with empty source info
+    parquet_reader_options options = parquet_reader_options::builder().filter(filter_expression);
+
+    // Fetch footer and page index bytes from the buffer.
+    auto const footer_buffer = fetch_footer_bytes(file_buffer_span);
+
+    // Create hybrid scan reader with footer bytes
+    auto const reader =
+      std::make_unique<parquet::experimental::hybrid_scan_reader>(footer_buffer, options);
+
+    // Get all row groups from the reader
+    auto input_row_group_indices = reader->all_row_groups(options);
+
+    // Span to track current row group indices
+    auto current_row_group_indices = cudf::host_span<cudf::size_type>(input_row_group_indices);
+
+    // Calling `filter_data_pages_with_stats` before setting up the page index should raise an
+    // error
+    EXPECT_THROW(std::ignore = reader->build_row_mask_with_page_index_stats(
+                   current_row_group_indices, options, stream, mr),
+                 std::runtime_error);
+
+    // Set up the page index
+    auto const page_index_byte_range = reader->page_index_byte_range();
+    auto const page_index_buffer = fetch_page_index_bytes(file_buffer_span, page_index_byte_range);
+    reader->setup_page_index(page_index_buffer);
+
+    // Filter the data pages with page index stats
+    auto const row_mask =
+      reader->build_row_mask_with_page_index_stats(current_row_group_indices, options, stream, mr);
+
+    auto const expected_num_rows = reader->total_rows_in_row_groups(current_row_group_indices);
+    EXPECT_EQ(row_mask->type().id(), cudf::type_id::BOOL8);
+    EXPECT_EQ(row_mask->size(), expected_num_rows);
+    EXPECT_EQ(row_mask->null_count(), 0);
+
+    // Copy the row mask to the host and count the number of surviving rows
+    auto const host_row_mask = cudf::detail::make_host_vector<bool>(
```

Comment:
Can the compiler infer this type from the first argument so you don't have to type it here?